dateparser/date.py

```diff
@@ -461,6 +463,19 @@ def date_strings():
                     if self._is_applicable_locale(locale, s):
                         yield locale
 
+        if self.detect_languages_func:
+            detected_languages = self.detect_languages_func(
+                date_string, settings=self._settings
+            )
+
+            if self._settings.LANGUAGE_DETECTION_STRICT_USE:
+                self.languages = map_languages(detected_languages)
+            else:
+                if self.languages:
+                    self.languages += map_languages(detected_languages) or self._settings.DEFAULT_LANGUAGES
```

Comment:
Following the DRY principle ("DON'T REPEAT YOURSELF"), what about doing `languages = map_languages(detected_languages) or self._settings.DEFAULT_LANGUAGES` to simplify a little the `if-else`? :)